### DISCUSSION TITLE
Reverse proxy Immich docker windows

### DISCUSSION THREAD

**Discussion by trovador01** [immich-app/immich#2722] [Q&A]
Can someone please explain how to mount immich with docker windows. With Ngnix I have the certificate obtained, through duckdns, but I don't know what parameters to tell immich when installing the .env and docker-compose.yml files in it.

**Comment by r01k** [comment]
I don't think you have to touch anything on the Docker side. Simply configure nginx as a reverse proxy in front of immich_proxy. I was doing this yesterday on a new setup, but later replaced nginx for Caddy. I only use Docker on Linux, btw. This was my working nginx config passing HTTPS on 28376 to immich_proxy HTTP on 2283:

`server {

    listen 28376  ssl;
    server_name immich-server.ddns.net;

    # SSL Configuration

    ssl_certificate /etc/ssl/immich/server_cert.pem;
    ssl_certificate_key /etc/ssl/immich/server_key.pem;
    ssl_session_cache  builtin:1000  shared:SSL:10m;
    ssl_protocols TLSv1.2 TLSv1.3;
    ssl_ciphers HIGH:!aNULL:!eNULL:!EXPORT:!CAMELLIA:!DES:!MD5:!PSK:!RC4;
    ssl_prefer_server_ciphers on;

    # Set the access log location

    access_log            /var/log/nginx/immich_access.log;
    error_log            /var/log/nginx/immich_error.log  debug;

    location / {

      # Set the proxy headers

      proxy_set_header        Host $host;
      proxy_set_header        X-Real-IP $remote_addr;
      proxy_set_header        X-Forwarded-For $proxy_add_x_forwarded_for;
      proxy_set_header        X-Forwarded-Proto $scheme;

      # Configure which address the request is proxied to

      proxy_pass          http://localhost:2283;
      proxy_read_timeout  90;

      # Security headers
      add_header Strict-Transport-Security "max-age=31536000; includeSubDomains; preload";
      add_header X-Frame-Options DENY;
      add_header X-Content-Type-Options nosniff;
      add_header X-XSS-Protection "1; mode=block";
      add_header Referrer-Policy "origin";
    }

}
`

**Comment by trovador01** [comment]
I can't get it to work, an explanatory video would be great.
I have managed to install both the Ngnix and Immich programs, but I can't get the proxy to work with certificates. I have to touch some configuration in the immich files when installing it (.env and docker-compose.yml)

**Reply by jrasm91** [reply]
You may want to look into using Nginx proxy manager, caddy, or traefik, which all have YouTube tutorial videos for how to configure them for use with SSL.

**Reply by sjayanna01** [reply]
@jrasm91  Do you mind sharing your immich and caddy docker compose and the caddyfile? I've setup portainer successfully with caddy and tailscale on Synology NAS, but struggling to setup immich and teslamate. Immich is currently returning blank white page with 400 error in the caddy logs.
